### PR TITLE
chore: adding icons on buttons

### PR DESCRIPTION
**Pull request by marceloFerreira90**
closes https://github.com/nearform/quantum/issues/216
closes  https://github.com/nearform/quantum/issues/258

Added the buttons for left and right side. 

- Selected
Screenshot 2024-01-09 at 10 08 10

- Icons
Screenshot 2024-01-09 at 10 08 20

### CODE REVIEW

File: src/colors/button.ts [hunk 1]
```diff
@@ -19,6 +19,10 @@ export const button = {
     disabled: {
       DEFAULT: colors.grey['50'],
       dark: background.subtle.dark
+    },
+    selected: {
+      DEFAILT: colors.grey['100'],
```
Comment (andreaforni) [inline src/colors/button.ts:24]: **typo**: there is a typo here and in all the other `selected` objects.  
```suggestion
      DEFAULT: colors.grey['100'],
```

File: src/components/Button/index.tsx [hunk 2]
```diff
@@ -99,24 +108,70 @@ const buttonVariants = cva(
...
+            {...props}
+          />
+          {rightSideChild?
+          
+          <div className={cn(rightSideVariants(), rightSideVariants)}>
+             {rightSideChild}
+          </div>
+           :<></>
+          }
+      </form>
```
Comment (andreaforni) [inline src/components/Button/index.tsx:170]: **suggestion:** You don't need a `form` element wrapping the button. A `button` element accepts arbitrary `children`, so I would:
- remove the `form`
- move the `leftSideChild` and `rightSideChild` inside the `<button>` 
Something like:
```suggestion
        <button
          className={cn(buttonVariants(), className)}
          ref={ref}
          {...props}
        >
          {leftSideChild ? (
            <div className={cn(leftSideVariants(), leftSideClassName)}>
              {leftSideChild}
            </div>
          ) : (
            <></>
          )}
          {children}
          {rightSideChild ? (
            <div className={cn(rightSideVariants(), rightSideVariants)}>
              {rightSideChild}
            </div>
          ) : (
            <></>
          )}
        </button>
```

File: src/colors/button.ts (hunk 1 above)
Comment (nickzafiropulos) [reply src/colors/button.ts:24]: All looks good, thanks

**Comment by andreaforni** [comment]
LGTM

**Pull request merged**